icssc/AntAlmanac
# Add Past Enrollment Popup

## Summary
- Now that the enrollment history endpoint is implemented, we can bring back and improve the Past Enrollment popup.
- Currently, the popup just displays the enrollment history of the most recent lecture section offered.

Demo:

https://github.com/icssc/AntAlmanac/assets/78244965/1b6bcee9-cc37-4057-a85f-2fea05f3b124

## TODO

- [X] Make a request to PeterPortal to get the enrollment history, and parse the response
- [X] Create the popup and display an enrollment history graph
- [x] Cache responses
- [x] Add ZotTracker link

## Test Plan
- Verify this works for classes in the most recent quarter (e.g. CS 132).
- Verify this works for classes that haven't been offered very recently (e.g. CS 122D from Spring 2023).
- Verify this works for classes that have never been offered before (force `queryEnrollmentHistory` to return null).
- Make sure it looks good on dark and light mode, as well as mobile.

## Issues

Closes #341 
Closes #182 

## Future Followup
- We should allow the user to see the enrollment history of any lecture that was previously offered: #830
